### PR TITLE
Find available slots in a day

### LINKED ISSUE #9: Check available time
As a nurse, I can check the available time of the doctor.

### LINKED ISSUE #21: Get available dates
As a nurse frequently getting appointment calls, I am able to easily get dates availability while on the phone, such that I can finish my work more quickly.

### PR DESCRIPTION
**Pull request by xz0127**
Create models of annotated points, time intervals, point type for scheduling method support.
Implement scheduling manager for finding available time slots.
Add avail command and command parser.
Implement scheduling method using line-sweeping algorithm.
Implement tests for AvailableCommand, AvailableCommandParser and TimeConversionUtil.
Implement tests for the models constructed.
Implement functions to get the next available slot. An available slot is assumed to have 1 hour long and is no earlier than the current time.
Display next available slot following the list of all available slots.

Sample flow:
1. Operation time is from 8 AM to 10 PM, we construct operation interval list[(480, 1320)]
2. Annotate 480 as TimeIntervalStart and 1320 as TimeIntervalEnd.
3. Appointment time is from 8AM - 9AM, 9PM-10PM, we construct a new appintment time interval list[(480, 540), (1260, 1320)]
4. Annotate 480 and 1260 as AppointmentStart, annotate 540, 1320 as AppointmentEnd within this interval list.
5. Apply line-sweeping algorithm to obtain the non overlapping intervals: [(540, 1260)]
6. Obtain available slots: 9AM - 9PM

closes #9 and closes #21

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AvailableCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    /**
+     * Parse the given {@code String} of arguments in the context of AvailableCommand
+     * and returns an AvailableCommand for execution.
+     * @param userInput full user input string
+     * @return an AvailableCommand object
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public AvailableCommand parse(String userInput) throws ParseException {
```
Comment (JinHao-L) [inline src/main/java/seedu/address/logic/parser/AvailableCommandParser.java]: Should it be `parses` instead?

File: src/main/java/seedu/address/model/ScheduleManager.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        }
+        return new TimeIntervalList(appointmentTimeIntervals);
+    }
+
+    /**
+     * Add a list of annotated points to the annotated point list based on
+     * the opening and closing time interval of the clinic.
+     *
+     * @param operationTimeIntervals list of time intervals of clinic operation time.
+     */
```
Comment (JinHao-L) [inline src/main/java/seedu/address/model/ScheduleManager.java:100]: This shd be `adds` instead

File: src/main/java/seedu/address/model/ScheduleManager.java [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+        for (TimeInterval timeInterval : availableTimeSlots.getTimeIntervals()) {
+            if (timeInterval.isAtLeastOneHour()) {
+                Time startTime = timeInterval.getStartTime();
+                Time endTime = new Time(startTime.getTime().plusHours(1));
+                TimeInterval nextInterval = new TimeInterval(startTime, endTime);
+                return nextInterval.toString();
+            }
+        }
+        return "Sorry we cannot find any slot available for 1 hour appointment.";
+    }
```
Comment (JinHao-L) [inline src/main/java/seedu/address/model/ScheduleManager.java:230]: Sorry just a bit curious about this function, this is currently not used right?

**Review by JinHao-L — APPROVED** [review]
Looks good! Good work! 👍 
Couldn't really find any issue so I made some minor comments on the JavaDocs, otherwise LGTM!

File: src/main/java/seedu/address/logic/parser/AvailableCommandParser.java (hunk 1 above)
Comment (xz0127) [reply src/main/java/seedu/address/logic/parser/AvailableCommandParser.java]: Thanks for pointing this out! 👍

File: src/main/java/seedu/address/model/ScheduleManager.java (hunk 2 above)
Comment (xz0127) [reply src/main/java/seedu/address/model/ScheduleManager.java:100]: Thanks for pointing this out! 👍

File: src/main/java/seedu/address/model/ScheduleManager.java (hunk 3 above)
Comment (xz0127) [reply src/main/java/seedu/address/model/ScheduleManager.java:230]: Yes it was not used at this stage but I have implemented it in the next commit.

**Review by JinHao-L — APPROVED** [review]
LGTM! 👍

**Pull request merged**